### DISCUSSION TITLE
When tls configured login fails with 401 invalid login

### DISCUSSION THREAD

**Discussion by maximilianpohle** [minio/minio#13031] [Q&A] [answered]
Hello,

I'm running minio in a docker environment. As soon I deploy the certificates for the https connection I'm unable to login to the minio.
Error message: {"code":401,"detailedMessage":"invalid Login","message":"invalid Login"}

If I delete the certificates and restart the container I'm able to login again with http.

Mounted volumes:
```
"/srv/minio/config:/root/.minio/:rw",
"/srv/minio/data:/data:rw"
```

docker inspect
```
...
"Cmd": [
                "server",
                "/data",
                "--console-address",
                ":9001",
                "--address",
                ":443"
            ],
"Image": "minio/minio"
```

Certificates:
```
$ ls -al /srv/minio/config/certs/
drwx------ 2 root      root      4096 Aug 23 09:09 CAs
-rw-rw-r-- 1 minio minio 1678 Aug 23 09:05 private.key
-rw-rw-r-- 1 minio minio 2077 Aug 23 09:05 public.crt
```

## Your Environment
* MinIo Version:  RELEASE.2021-08-20T18-32-01Z
* Docker Version: 20.10.8

**Comment by harshavardhana** [comment, marked as answer]
> $ ls -al /srv/minio/config/certs/
> drwx------ 2 root      root      4096 Aug 23 09:09 CAs
> -rw-rw-r-- 1 minio minio 1678 Aug 23 09:05 private.key
> -rw-rw-r-- 1 minio minio 2077 Aug 23 09:05 public.crt

Put the `public.crt` in to `CAs` folder and also read the docs

https://docs.min.io/minio/baremetal/reference/minio-server/minio-server.html#minio-console

```
MINIO_SERVER_URL
Optional

Specify the URL hostname the MinIO Console should use for connecting to the MinIO Server.

This variable may be necessary if the MinIO Server TLS certificates do not 
contain any IP Subject Alternative Names (SAN). Specifically, the Console 
uses the MinIO Server IP address by default. If the Server TLS does not 
contain that IP address, then the Console cannot validate the TLS connection.
```

**Comment by liyin37** [comment]
docker run -it -d  -p 8443:8443   -p 443:443   --name minio1    -v /mnt/minio:/root/.minio -v /mnt/data:/data   -e "MINIO_ROOT_USER=admin"   -e "MINIO_ROOT_PASSWORD=xxxxx"  -e "MINIO_DOMAIN=gisuni.xyz" -e "MINIO_SERVER_URL=https://gisuni.xyz"  minio/minio server /data --console-address ":8443" --address ":443"

![image](https://user-images.githubusercontent.com/42507175/135242347-fea2eb9b-14b8-48d8-8cd6-24e9f9508e56.png)
[root@admin /]# tree /mnt/minio/
/mnt/minio/
└── certs
    ├── CAs
    │   └── public.crt
    ├── private.key
    └── public.crt

2 directories, 3 files
